src/app/src/sensorUtils.js

```diff
@@ -17,46 +29,87 @@ export function makeRiverGaugeRequest(id, isApiRequest) {
+    const data = Papa.parse(csvString, {
+        header: true,
+        comments: '#',
+        dynamicTyping: true,
+        skipEmptyLines: true,
+    });
+
+    return data.data;
 }
 
-export function parseRiverGaugeApiData(id, apiData) {
-    const {
-        data: {
-            value: {
-                timeSeries: data,
-            },
+export function parseRiverGaugeApiData(id, data) {
+    const extractedVariableData = VARIABLES.reduce(
+        (acc, variable, idx) => {
+            const apiVariableData = data[idx];
+            if (apiVariableData) {
+                const sensorValue = Number(
+                    apiVariableData.values[0].value[0].value
+                );
+                return sensorValue !== apiVariableData.variable.noDataValue
+                    ? Object.assign(acc, { [variable]: sensorValue })
+                    : Object.assign(acc, { [variable]: 0 });
+            }
+            return acc;
         },
-    } = apiData;
-
-    const extractedVariableData = VARIABLES.reduce((acc, variable, idx) => {
-        const apiVariableData = data[idx];
-        if (apiVariableData) {
-            const sensorValue = Number(apiVariableData.values[0].value[0].value);
-            return sensorValue !== apiVariableData.variable.noDataValue
-                ? Object.assign(acc, {[variable]: sensorValue })
-                : Object.assign(acc, {[variable]: 0 });
-        }
-        return acc;
-    }, { id });
+        { id }
+    );
 
     return extractedVariableData;
 }
 
-export function parseRiverGaugeCsvData(id, csvString) {
-    const {
-        data,
-    } = csvString;
-
-    const parsedData = parseCsvString(data);
-    const dataRow = parsedData.data.slice(-1)[0];
+export function parseRiverGaugeCsvData(id, data) {
+    const dataRow = data.slice(-1)[0];
 
-    const extractedVariableData = VARIABLES.reduce((acc, variable) => {
-        const code = `p${VARIABLE_CODES[variable]}`;
-        return Object.assign(acc, {[variable]: dataRow[code] || 0 });
-    }, { id });
+    const extractedVariableData = VARIABLES.reduce(
+        (acc, variable) => {
+            const code = `p${VARIABLE_CODES[variable]}`;
+            return Object.assign(acc, { [variable]: dataRow[code] || 0 });
+        },
+        { id }
+    );
 
     return extractedVariableData;
 }
+
+export function getSensorByProp(prop, value) {
+    const sensor = find(sensors.features, f => {
+        return f.properties[prop] === value;
+    });
+
+    return sensor;
+}
+
+export function calculateOverallSensorRating(sensorRatings) {
+    const totalValue = values(sensorRatings).reduce((acc, r) => acc + r, 0);
+
+    if (totalValue >= 4) {
+        return RATING_GOOD;
+    } else if (totalValue >= 2) {
+        return RATING_FAIR;
+    } else {
+        return RATING_POOR;
+    }
+}
+
+export function transformSensorDataToRatings(sensorData) {
+    const sensor = getSensorByProp('Id', sensorData.id).properties;
```

Comment:
Nitpick but this could be

```
const { properties: sensor } = getSensorByProp('Id', sensorData.id);
```

May also be worth setting `"Id"` to a constant.

Reply:
Thanks. Re: `Id`, I was planning on opening a clean-up issue related to our use of `id`. I'll add this to the list.

Reply:
https://github.com/azavea/ism-watershed-wellness-snapshot/issues/66